### PR TITLE
Faster RCNN for Object Detection - Full-Precision, No Compression

### PR DESCRIPTION
**Pull request by jyhengcoder**
add code from https://github.com/DetectionTeamUCAS/Faster-RCNN_Tensorflow

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/PocketFlow?pullRequest=246) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/PocketFlow?pullRequest=246) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/PocketFlow?pullRequest=246) it.</sub>

File: datasets/pascalvoc_dataset.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
 tf.app.flags.DEFINE_integer('nb_smpls_train', 22136, '# of samples for training')
 tf.app.flags.DEFINE_integer('nb_smpls_val', 500, '# of samples for validation')
 tf.app.flags.DEFINE_integer('nb_smpls_eval', 4952, '# of samples for evaluation')
-tf.app.flags.DEFINE_integer('batch_size', 32, 'batch size per GPU for training')
+tf.app.flags.DEFINE_integer('batch_size', 1, 'batch size per GPU for training')
```
[the remark this answers is not in the document]
Comment (jiaxiang-wu) [reply datasets/pascalvoc_dataset.py]: Resolved in cff5a1d.

File: learners/abstract_learner.py [hunk 2]
```diff
@@ -84,6 +84,7 @@ def __init__(self, sm_writer, model_helper):
     self.dump_n_eval = model_helper.dump_n_eval
     self.model_name = model_helper.model_name
     self.dataset_name = model_helper.dataset_name
+    self.enbl_label = model_helper.enbl_label
```
[the remark this answers is not in the document]
Comment (jiaxiang-wu) [reply learners/abstract_learner.py]: Resolved in cff5a1d.

File: learners/full_precision/learner.py [hunk 3]
```diff
@@ -131,7 +131,10 @@ def __build(self, is_train):  # pylint: disable=too-many-locals
       # model definition - primary model
       with tf.variable_scope(self.model_scope):
         # forward pass
-        logits = self.forward_train(images) if is_train else self.forward_eval(images)
+        if self.enbl_label:
```
[the remark this answers is not in the document]
Comment (jiaxiang-wu) [reply learners/full_precision/learner.py]: Resolved in cff5a1d.

File: nets/abstract_model_helper.py [hunk 4]
```diff
@@ -21,33 +21,28 @@
 
 class AbstractModelHelper(ABC):
   """Abstract class for model helpers.
-
```
[the remark this answers is not in the document]
Comment (jiaxiang-wu) [reply nets/abstract_model_helper.py]: Resolved in cff5a1d.

File: nets/abstract_model_helper.py [hunk 5]
```diff
@@ -145,4 +129,4 @@ def model_name(self):
   @abstractmethod
   def dataset_name(self):
     """Dataset's name."""
-    pass
+    pass
```
[the remark this answers is not in the document]
Comment (jiaxiang-wu) [reply nets/abstract_model_helper.py]: Resolved in cff5a1d.

File: nets/abstract_model_helper.py [hunk 6]
```diff
@@ -21,33 +21,28 @@
...
   A model helper should define the following function interface:
     1. Data input pipeline for training and evaluation subsets.
     2. Network's forward pass during training and evaluation.
     3. Loss function (and some extra evaluation metrics).
-
   All functions marked with "@abstractmethod" must be explicitly implemented in the sub-class.
   """
 
-  def __init__(self, data_format):
+  def __init__(self, data_format="channels_last"):
```
[the remark this answers is not in the document]
Comment (jiaxiang-wu) [reply nets/abstract_model_helper.py]: Resolved in c5953bc.

File: datasets/pascalvoc_dataset.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiaxiang-wu) [reply datasets/pascalvoc_dataset.py]: Resolved in b79833e.

**Pull request merged**